tests/neg/i15991.abstract.scala

```diff
@@ -0,0 +1,20 @@
+object Foo:
+  def unapply[T <: Tuple](tup: T): String *: String *: T =
+    "a" *: "b" *: tup
+
+// like {pos,neg}/i15991, but with an abstract tuple tail
+class Test:
+  val tup2: String *: String *: EmptyTuple = ("c", "d")
+
+  def test3 =
+    val Foo(x, y, z) = tup2 // error: Wrong number of argument patterns for Foo; expected: (String, String)
```

Comment:
Why does the error say here that the pattern expects `(String, String)` – only two arguments and not four?

Reply:
Foo only returns two strings so trying to extract 3 is wrong.

Reply:
Foo returns two strings plus a tuple that it extracts, no? So that makes for more than two arguments (unless the unapplied tuple is empty)?

Reply:
You're right, well spotted.  I'll go back and check why we're saying expected 2 instead of 4.

Reply:
LOL, I misread it as well, repeatedly

Reply:
Oh, I just had an older copy of the error message, from when 2 was the correct number.